Link error, undefined symbol: PcgRandom::PcgRandom

##### Minetest version
<!--
Paste Minetest version between quotes below
If you are on a devel version, please add git commit hash
You can use `minetest --version` to find it.
-->
```
Commit hash 4c7623981 from the master branch
```

##### OS / Hardware
<!-- General information about your hardware and operating system -->
Operating system: macOS Catalina 10.15.7 with XCode 12.3
CPU: 2.4GHz Intel Core 2 Duo (my machine is a MacBook7,1)

<!-- For graphical issues only -->
GPU model:
OpenGL version:

##### Summary
<!-- Describe your problem here -->
Upon reaching the linking phase CMake prints out this error:
```
[ 27%] Linking CXX executable ../../../bin/minetest
Undefined symbols for architecture x86_64:
  "PcgRandom::PcgRandom(unsigned long long, unsigned long long)", referenced from:
      Sky::updateStars() in sky.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [../../bin/minetest] Error 1
make[1]: *** [src/CMakeFiles/minetest.dir/all] Error 2
make: *** [all] Error 2
```
(the build is in `<repo root>/build/macOS/`)
##### Steps to reproduce
<!-- For bug reports or build issues, explain how the problem happened -->
1. Clone repository
2. Create `build/macOS` directory and `cd` into it
3. Call cmake to generate make stuff
 * `cmake ../.. -DBUILD_CLIENT=1 -DBUILD_SERVER=0 -DENABLE_FREETYPE=1 -DCMAKE_EXE_LINKER_FLAGS='-L/usr/local/opt/freetype/lib' -DENABLE_GETTEXT=1 -DCUSTOM_GETTEXT_PATH=/usr/local/opt/gettext`
4. Call make to compile
 * `make -j4`
5. Build compiles successfully but errors out at the Link stage.

Notes:

Additionally here are [CMakeOutput.log](https://privatebin.mentalabs.eu/?33bc62ed146b5fbf#PcTVtbttXUo88MjNSKSiGbkuJcvSVa4bRQDJjQKW9hW) and [CMakeError.log](https://privatebin.mentalabs.eu/?8347deb46843b809#E8nDiu6jWhquZGntCEkQeMvabMuCBmgtJ8Uq795NwpSW).

I used the `cmake` command that the Homebrew formula uses as I thought that was going to work.

--- a/src/client/fontengine.h
+++ b/src/client/fontengine.h
@@ -22,6 +22,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <map>
 #include <vector>
 #include "util/basic_macros.h"
+#include "irrlichttypes.h"
 #include <IGUIFont.h>
 #include <IGUISkin.h>
 #include <IGUIEnvironment.h>

--- a/src/client/gameui.h
+++ b/src/client/gameui.h
@@ -20,6 +20,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
+#include "irrlichttypes.h"
 #include <IGUIEnvironment.h>
 #include "gui/guiFormSpecMenu.h"
 #include "util/enriched_string.h"

--- a/src/client/shader.h
+++ b/src/client/shader.h
@@ -20,8 +20,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
-#include <IMaterialRendererServices.h>
 #include "irrlichttypes_bloated.h"
+#include <IMaterialRendererServices.h>
 #include <string>
 #include "tile.h"
 #include "nodedef.h"

--- a/src/client/sky.h
+++ b/src/client/sky.h
@@ -17,10 +17,10 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 */
 
+#include "irrlichttypes_extrabloated.h"
 #include <ISceneNode.h>
 #include <array>
 #include "camera.h"
-#include "irrlichttypes_extrabloated.h"
 #include "irr_ptr.h"
 #include "shader.h"
 #include "skyparams.h"

--- a/src/gui/guiEditBox.h
+++ b/src/gui/guiEditBox.h
@@ -19,6 +19,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
+#include "irrlichttypes.h"
 #include "IGUIEditBox.h"
 #include "IOSOperator.h"
 #include "guiScrollBar.h"

--- a/src/gui/touchscreengui.h
+++ b/src/gui/touchscreengui.h
@@ -18,6 +18,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
+#include "irrlichttypes.h"
 #include <IEventReceiver.h>
 #include <IGUIButton.h>
 #include <IGUIEnvironment.h>

